app/lib/frontend/name_tracker.dart

```diff
@@ -91,27 +95,49 @@ class NameTrackerUpdater {
     _logger.info(
         'Scanned initial package names (${nameTracker.length}) in ${sw.elapsed}.');
 
-    await Future.delayed(_pollingInterval);
+    await _sleep();
 
     _logger.info('Monitoring new package creation.');
     for (;;) {
+      if (_stopped) return;
       try {
         await _scan();
       } catch (e, st) {
         _logger.severe(e, st);
       }
-      await Future.delayed(_pollingInterval);
+      await _sleep();
     }
   }
 
+  Future _sleep() async {
+    _sleepCompleter = Completer();
```

Comment:
Since you can only ever stop once... `stop = false` is never flipped back, I would suggest creating a single `_sleepCompleter` at the start.. and maybe call it `stopping`...

Note: this is not important.